pkg/operators/eventgen/dns.go

```diff
@@ -0,0 +1,123 @@
+package eventgen
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/inspektor-gadget/inspektor-gadget/pkg/logger"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/kubernetes/scheme"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/remotecommand"
+)
+
+type DNSGenerator struct {
+    clientset *kubernetes.Clientset
+    config    *rest.Config
+    logger    logger.Logger
+}
+
+func NewDNSGenerator(config *rest.Config, log logger.Logger) (Generator, error) {
+    clientset, err := kubernetes.NewForConfig(config)
+    if err != nil {
+        return nil, fmt.Errorf("failed to create Kubernetes client: %v", err)
+    }
+    return &DNSGenerator{
+        clientset: clientset,
+        config:    config,
+        logger:    log, 
+    }, nil
+}
+
+func (d *DNSGenerator) Generate(domain string) (string, string, string, error) {
+    namespace := "default"
+    podName := fmt.Sprintf("dns-test-pod-%d", time.Now().Unix())
+
+    container, err := d.createAndWaitForPod(namespace, podName)
+    if err != nil {
+        return "", "", "", err
+    }
+
+    if err := d.executeDNSLookup(namespace, podName, domain); err != nil {
+        return "", "", "", err
+    }
```

Comment:
I'm thinking how we can make these more generic (using an interface) to generate events on k8s and non-k8s containers. Unfortunately, I noticed [container-utils/testutils](https://pkg.go.dev/github.com/inspektor-gadget/inspektor-gadget/pkg/container-utils/testutils) package can't be used as it's thought to be used on tests, so the whole implementation depends on the [testing](https://pkg.go.dev/testing) package.

I think we can make `container-utils/testutils` independent from testing, but the k8s implementation requires a lot of work. So, I'm wondering what other options we have.

cc @mauriciovasquezbernal